docs/data/date-pickers/date-picker/CustomDay.tsx

```diff
@@ -63,12 +63,20 @@ function Day(props: PickersDayProps<Dayjs>) {
 }
 
 export default function CustomDay() {
+  const [value, setValue] = React.useState<Dayjs | null>(dayjs('2022-04-07'));
+
   return (
     <LocalizationProvider dateAdapter={AdapterDayjs}>
       <StaticNextDatePicker
         displayStaticWrapperAs="desktop"
-        defaultValue={dayjs('2022-04-07')}
+        value={value}
+        onChange={(newValue) => setValue(newValue)}
         components={{ Day }}
+        componentsProps={{
+          day: {
+            selectedDay: value,
+          } as any,
```

Comment:
To do avoid the case, users would have to use the slot props override interfaces, 
But inside our doc I don't know how it would work exactly. 
I will create the override interface and extend it in this doc examples to see how it behaves.